internal/controller/driver_controller.go

```diff
@@ -226,6 +226,10 @@ func (r *DriverReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 func (r *driverReconcile) reconcile() error {
 	// Load the driver desired state based on driver resource, operator config resource and default values.
 	if err := r.LoadAndValidateDesiredState(); err != nil {
+		if k8serrors.IsNotFound(err) {
+			r.log.Info("Driver resource does not exist anymore, skipping reconcile")
+			return nil
+		}
```

Comment:
I think this is should not be here because it is returning nil for all the not found errors, like if the imageset configmap not found we will get here and we return nil to stop the reconcile which is not correct.

Reply:
I agree.
The inner function needs to return a different custom error so the top-most function can make a decision based on that concrete error

Reply:
agreed, done.